### PR TITLE
Add warning on unset edition

### LINKED ISSUE #1662: Add a warning on unset edition
emit a warning if edition is not set 
https://github.com/software-mansion/scarb/blob/main/scarb/src/core/manifest/toml_manifest.rs#L585

### PR DESCRIPTION
**Pull request by DelevoXDG**
Closes #1662

### CODE REVIEW

File: scarb/src/core/manifest/toml_manifest.rs [hunk 1]
```diff
@@ -582,7 +583,10 @@ impl TomlManifest {
             .clone()
             .map(|edition| edition.resolve("edition", || inheritable_package.edition()))
             .transpose()?
-            .unwrap_or_default();
+            .unwrap_or_else(|| {
+                config.ui().warn("`edition` not found in 'package' section");
```
Comment (maciektr) [inline scarb/src/core/manifest/toml_manifest.rs:587]:
```suggestion
                config.ui().warn("`edition` field not set in `[package]` section");
```

File: scarb/src/core/manifest/toml_manifest.rs
Comment (maciektr) [inline scarb/src/core/manifest/toml_manifest.rs:1]: per offline discussion - this warning should not be shown for `cairo-plugin` packages

File: scarb/src/core/manifest/toml_manifest.rs (hunk 1 above)
Comment (DelevoXDG) [reply scarb/src/core/manifest/toml_manifest.rs:587]: Maybe we should also include the package id, WDYT?

```suggestion
                config.ui().warn(format!("`edition` field not set in `[package]` section for package {package_id}"));
```

Comment (maciektr) [reply scarb/src/core/manifest/toml_manifest.rs]: Name would be enough IMHO

**Review by maciektr — APPROVED** [review]

File: scarb/tests/build_cairo_plugin.rs [hunk 2]
```diff
@@ -194,6 +194,7 @@ fn can_emit_plugin_warning() {
     ProjectBuilder::start()
         .name("hello")
         .version("1.0.0")
+        .edition("2023_01")
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply scarb/tests/build_cairo_plugin.rs]: Yep, you're right. Removed 👍🏻

File: scarb/tests/fmt.rs [hunk 3]
```diff
@@ -143,7 +146,10 @@ fn simple_format_with_filter() {
         .current_dir(&t)
         .assert()
         .failure()
-        .stdout_eq("error: package `world` not found in workspace\n");
+        .stdout_eq(indoc! {r#"
+            warn: `edition` field not set in `[package]` section
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply scarb/tests/fmt.rs]: Yeah, addressed prior to you publishing the review 😅 [7b3e493](https://github.com/software-mansion/scarb/pull/1663/commits/7b3e49349b99ff37383e050d574c3a0dc2fa1a3e)

**Pull request merged**